libBlueSky/SizeGrip.cpp

```diff
@@ -0,0 +1,224 @@
+#include "SizeGrip.hpp"
+
+#include <QAbstractScrollArea>
+#include <QApplication>
+#include <QDesktopWidget>
+#include <QLayout>
+#include <QMouseEvent>
+
+namespace BlueSky {
+
+SizeGrip::SizeGrip(QWidget* parent)
+    : QSizeGrip(parent)
+    , mResizeDirections(ResizeDefault)
+    , mParentWindow(parent)
+    , mGotMousePress(false)
+    , dxMax(0)
+    , dyMax(0)
+    , mCorner(mParentWindow->isLeftToRight() ? Qt::BottomRightCorner : Qt::BottomLeftCorner)
+{
+    Q_ASSERT(parent);
+    resize(sizeHint());
+}
+
+SizeGrip::ResizeDirections SizeGrip::resizeBehaviour() const
+{
+    return mResizeDirections;
+}
+
+void SizeGrip::setResizeBehaviour(int behaviour)
+{
+    mResizeDirections = static_cast<ResizeDirections>(behaviour);
+}
+
+void SizeGrip::mousePressEvent(QMouseEvent* ev)
+{
+    if (ev->button() != Qt::LeftButton) {
+        QWidget::mousePressEvent(ev);
+        return;
+    }
+
+    QWidget *tlw = mParentWindow;
+    mGotMousePress = true;
+    mStartPos = ev->globalPos();
+    mDlgRect = mParentWindow->geometry();
+
+#ifdef Q_WS_X11
```

Comment:
Yeah, this happens if you copy 3rd party code. Unfortunately we don't get information about event handling from the base implementation. So I don't see a chance currently here. Maybe you have a better idea?